build.gradle

```diff
@@ -208,6 +209,55 @@ ext {
 
 ext['snakeyaml.version'] = '2.0'
 
+//here we are going to store swagger files
+def swaggerList = []
+
+//iteration by swagger file root folder and save into swaggerList variable
+def dir = new File("$rootDir/src/main/resources/openapi/".toString())
+import static groovy.io.FileType.FILES
+
+dir.eachFileRecurse(FILES) {file ->
+  if (file.getName().endsWith(".yaml"))
+    swaggerList << file
+}
+
+def openApiGenerateTaskList = []
+
+// Iterate on all swagger files and generate a task for each one with the nomenclature openApiGenerate + swagger name
+swaggerList.each {
+  def apiName = it.getName().replace(".yaml", "");
+  def taskName = "openApiGenerate" + apiName.capitalize()
+  openApiGenerateTaskList << taskName
+  tasks.create(taskName, org.openapitools.generator.gradle.plugin.tasks.GenerateTask, {
+    generatorName = "java"
+    inputSpec = "$rootDir/src/main/resources/openapi/".toString() + "${apiName}.yaml"
+    outputDir = "$buildDir/generated/openapi".toString()
+    modelPackage = "uk.gov.hmcts.darts.model.".toString() + "${apiName}"
+    //    https://openapi-generator.tech/docs/generators/spring/#config-options
+    skipOperationExample = true
+    skipValidateSpec = true
+    configOptions = [
+      dateLibrary   : "java8",
+      serializationLibrary       : "gson",
```

Comment:
so does this change the annotations generated from Jackson to Gson?

Reply:
yup